### ISSUE TITLE
Add a build step at the end of .travis.yml that uploads working wheels to a known location

### ISSUE THREAD

**Issue by njsmith** [scipy/scipy#5379]
There are projects that want to run their tests against the latest scipy master, and it would be a lot easier for them if there was a wheel they could just download. Some of them do `pip install git+http://...path/to/scipy/master`, but this has two disadvantages: (a) everyone has to rebuild from scratch all the time, (b) sometimes (hopefully rarely, but...) master is broken and this just doesn't work.

The solution is that we should have an `after_success:` block in .travis.yml so that after each commit has passed its tests, we upload the corresponding wheel somewhere, probably one of these rackspace accounts that everyone seems to have access to.

(Ideally we would only do this if *all* tests pass, but unfortunately this is impossible with travis currently: https://github.com/travis-ci/travis-ci/issues/929 . In the mean time, uploading each build variant iff its tests pass seems like a reasonable compromise.)

@ogrisel volunteers to do this in a few weeks, so if it's, say, mid-November and it hasn't happened, then someone should nag him ;-)
(https://github.com/numpy/numpy/issues/6462#issuecomment-148167879)

(See also numpy/numpy#6493)

**Comment by rgommers** [comment]
good idea. marking this for 0.17.0 so we get a reminder of this issue in November.

**Comment by rgommers** [comment]
I won't nag, but here's a friendly reminder @ogrisel:)

**Comment by ogrisel** [comment]
I am back from vacation. I will try to start with numpy first.

**Comment by ogrisel** [comment]
@njsmith this issue is tackled in #5609.

**Comment by ogrisel** [comment]
Ok the merge triggered a build that successfully uploaded the wheel to:

http://travis-dev-wheels.scipy.org/scipy-0.18.0.dev0+20151229143833_eecc0ed-cp35-cp35m-linux_x86_64.whl

as per travis build log:

```
Building wheels for collected packages: futures, backports.ssl-match-hostname
  Running setup.py bdist_wheel for futures
  Stored in directory: /home/travis/.cache/pip/wheels/66/53/5c/436e13e3ab70dc8fce6d0c9e1715d52378a72472053eb7efc0
  Running setup.py bdist_wheel for backports.ssl-match-hostname
  Stored in directory: /home/travis/.cache/pip/wheels/ca/e4/ff/d57f7b2052ddfcf76644f59ad355808f8584d79d4d922601e1
Successfully built futures backports.ssl-match-hostname
Installing collected packages: packaging, futures, backports.ssl-match-hostname, apache-libcloud, wheelhouse-uploader
Successfully installed apache-libcloud-0.20.0 backports.ssl-match-hostname-3.5.0.1 futures-3.0.3 packaging-15.3 wheelhouse-uploader-0.9.1
Renaming dev wheel to add an upload timestamp: scipy-0.18.0.dev0+20151229143833_eecc0ed-cp35-cp35m-linux_x86_64.whl
About to upload 1 files
Uploading /home/travis/build/scipy/scipy/wheelhouse/scipy-0.18.0.dev0+20151229143833_eecc0ed-cp35-cp35m-linux_x86_64.whl [29.196 MB]
Uploading metadata.json
Updating index.html with 5 links
Wheelhouse successfully published at:
http://66ce22ef0bf26a5e2cca-4ffdece11fd3f72855e4665bc61c7445.r49.cf2.rackcdn.com
travis_time:end:0b26d7ad:start=1451399908251290126,finish=1451399923447495650,duration=15196205524
```

The metadata file also has the new entry for the scipy wheel:

http://travis-dev-wheels.scipy.org/metadata.json

```
{"numpy-1.11.0.dev0+20151229093430_004639d-cp35-cp35m-linux_x86_64.whl": {"sha256": "311dbdcc6a20f2ea4777ca7f66c0643c8095541c8e10aec37717d3122ba5b52b", "size": 5745622}, "scipy-0.18.0.dev0+20151229143833_eecc0ed-cp35-cp35m-linux_x86_64.whl": {"sha256": "c5f4c8da41abd6c5f478d1d568bc1c0c9eea390009e41dd862503412b5a952b2", "size": 29195737}, "numpy-1.11.0.dev0+20151228223537_efd5fbe-cp35-cp35m-linux_x86_64.whl": {"sha256": "7998eac0837638b635f209dc1267debae97ab24ee02ada9c0365b53876b69319", "size": 5745621}, "numpy-1.11.0.dev0+20151226190554_c9a8304-cp35-cp35m-linux_x86_64.whl": {"sha256": "9f2452652ce846a52fb5f55095c5e5abc27bb27ea9b198c51816eb72de81649e", "size": 5744141}, "numpy-1.11.0.dev0+20151226235450_106bfd6-cp35-cp35m-linux_x86_64.whl": {"sha256": "dea047cd22e204653f133cdf86d97510b7e0c9c212b1467fe292e7488dbc024a", "size": 5744118}, "numpy-1.11.0.dev0+20151228185836_ecbcbe0-cp35-cp35m-linux_x86_64.whl": {"sha256": "317bbc928c5f3f11abbc96613b5042eedec83cab828d7502d995ee46a1e41053", "size": 5744134}}
```

But for some reason it the scipy link was not added to the html index for some reason:

```
Updating index.html with 5 links
```

those 5 links are the last 5 numpy wheels links. I will investigate in the code of wheelhouse_uploader.

**Comment by ogrisel** [comment]
The index.html update issue is now fixed in wheelhouse_uploader 0.9.2 I just pushed to PyPI.

https://travis-dev-wheels.scipy.org is now up to date!

I think we can close this issue.

**Comment by rgommers** [comment]
Closing, thanks!
